pandas/_libs/window/aggregations.pyx

```diff
@@ -376,9 +385,13 @@ def roll_var(const float64_t[:] values, ndarray[int64_t] start,
             # never removed
             if i == 0 or not is_monotonic_increasing_bounds or s >= end[i - 1]:
 
+                prev_value = values[s]
+                num_consecutive_same_value = 0
```

Comment:
This can be set in `cdef`

Reply:
Should them be reset when `not is_monotonic_increasing_bounds or s >= end[i - 1]`? I don't quite understand when will these conditions be true, so I moved them from cdef to here.

Reply:
I think we support rolling if the index is monotonically decreasing for example (might be good to add a test for that).

Overall, it seem like `num_consecutive_same_value = 0` will always be set (because `i==0`) so better to add it in the cdef?

Reply:
So is it a feature that will be supported in the future?

 If that's the case, the count should be __reset__ in every window or it'll be messed up.

 For current use case where `not is_monotonic_increasing_bounds or s >= end[i - 1]` is always not True, we can move it to cdef.

Reply:
I do see we have some tests with monotonically decreasing index so it appears supported

`test_rolling_on_decreasing_index`
`test_rolling_decreasing_indices`
`test_rolling_decreasing_indices_centered`

Reply:
I've read these tests. Though they're named _decreasing_index_, they are still computed from top to bottom. It's just the __Index__ that is inverse.

For example in `test_rolling_on_decreasing_index`:
https://github.com/pandas-dev/pandas/blob/9a2d9ea214ea65ad93149c4987260254b5fe1329/pandas/tests/window/test_timeseries_window.py#L731-L744
`window_indexer.get_window_bounds` will return:
![image](https://user-images.githubusercontent.com/48577571/160401645-6744d8d7-c713-401a-8577-19d25df841bc.png)
these indices are still mono-increasing.

Then back to our case in `roll_var`, what I'm thinking is that, if we have real use case in the future that we actully compute from the last row up to the first row (mono-decreasing bounds), we should place reset in the if block. Otherwise if we can make sure all rolling algos is from top to bottom, we can place it to cdef.

Anyway, it should be safe to place it to cdef for now. Cuz if there are mono-decr bounds, the `roll_var` algo is false too. It should clearly have to remove obs from the tail and this logic is not present yet.

Reply:
`prev_value` can't be set in cdef. If length is 0, it'll cause a segfault.

Reply:
It's so strange that this version will casuse the same error as I used Py_ssize_t for count type in 32 bit test.
![47d5e970832424a85c6d387e22bbdde](https://user-images.githubusercontent.com/48577571/160422496-8f1d1d28-0492-4c17-8923-0e3a0209d5a8.png)
![6cead89f87de3fe11b311cf44f05fa0](https://user-images.githubusercontent.com/48577571/160422505-98f6fb6d-dd58-4be8-93b4-2063170f9487.png)
Revert it back now.

Reply:
Ah I see what you mean now. Yes, rolling will never be evaluated "bottom" to "top" though the index may be ordered in a monotonically decreasing fashion.